src/Streamly/Internal/Prelude.hs

```diff
@@ -3138,25 +3142,72 @@ concatMapTreeYieldLeavesWith combine f = concatMapLoopWith combine f yield
 ------------------------------------------------------------------------------
 
 -- Splitting operations that take a predicate and a Fold can be
--- expressed using splitParse. Operations like chunksOf, intervalsOf, split*,
--- can be expressed using splitParse when used with an appropriate Parse.
+-- expressed using parseMany. Operations like chunksOf, intervalsOf, split*,
+-- can be expressed using parseMany when used with an appropriate Parser.
 --
--- | Apply a 'Parse' repeatedly on a stream and emit the parsed values in the
+-- XXX We need takeGE/takeBetween to implement "some" using "many".
+
+-- | Apply a 'Parser' repeatedly on a stream and emit the parsed values in the
 -- output stream.
 --
--- >>> S.toList $ S.splitParse (PR.take 2 $ PR.fromFold FL.sum) $ S.fromList [1..10]
+-- This is the streaming equivalent of the 'Streamly.Internal.Data.Parser.many'
+-- parse combinator.
+--
+-- >>> S.toList $ S.parseMany (PR.take 2 $ PR.fromFold FL.sum) $ S.fromList [1..10]
 -- > [3,7,11,15,19]
 --
--- >>> S.toList $ S.splitParse (PR.line FL.toList) $ S.fromList "hello\nworld"
+-- >>> S.toList $ S.parseMany (PR.line FL.toList) $ S.fromList "hello\nworld"
 -- > ["hello\n","world"]
 --
-{-# INLINE splitParse #-}
-splitParse
+-- /Internal
+--
+{-# INLINE parseMany #-}
+parseMany
+    :: (IsStream t, MonadThrow m)
+    => Parser m a b
+    -> t m a
+    -> t m b
+parseMany p m =
+    D.fromStreamD $ D.parseMany (PRD.fromParserK p) (D.toStreamD m)
+
+-- | @parseManyTill collect test stream@ tries the parser @test@ on the input,
+-- if @test@ fails it backtracks and tries @collect@, after @collect@ succeeds
+-- @test@ is tried again and so on. The parser stops when @test@ succeeds.  The
+-- output of @test@ is discarded and the output of @collect@ is emitted in the
+-- output stream. The parser fails if @collect@ fails.
+--
+-- /Unimplemented/
+--
+{-# INLINE parseManyTill #-}
+parseManyTill ::
+    -- (IsStream t, MonadThrow m) =>
+       Parser m a b
+    -> Parser m a x
+    -> t m a
+    -> t m b
+parseManyTill = undefined
+
+-- | Iterate a parser generating function on a stream. The initial value @b@ is
+-- used to generate the first parser, the parser is applied on the stream and
+-- the result is used generate the next parser and so on.
+--
+-- >>> S.toList $ S.parseIterate (\b -> PR.take 2 (FL.mconcatTo b)) 0 $ S.fromList [1..10]
```

Comment:
Integers do not have a `Monoid` instance AFAIK

Reply:
fixed the example.